include/onnxruntime/core/platform/threadpool.h

```diff
@@ -138,6 +146,13 @@ class ThreadPool {
   // Schedules fn() for execution in the pool of threads.
   void Schedule(std::function<void()> fn);
 
+  // Run fn with up to n degree-of-parallelism enlisting the thread pool for
+  // help.  The degree-of-parallelism includes the caller, and so if n==1
+  // then the function will run directly in the caller.  The fork-join 
+  // synchronization is handled in the thread pool, and so any state captured
+  // by fn() is safe from concurrent access once RunWithHelp returns.
+  void RunWithHelp(std::function<void()> fn, int n);
```

Comment:
Is ExecuteInParallel or RunInParallel a better name? Also, should this be public?